kitsune/wiki/utils.py

```diff
@@ -182,26 +182,53 @@ def get_visible_document_or_404(
     except Document.DoesNotExist:
         pass
 
-    if (
-        not look_for_translation_via_parent
-        or not (locale := kwargs.get("locale"))
-        or (locale == settings.WIKI_DEFAULT_LANGUAGE)
-    ):
-        # We either don't want to try to find the translation via its parent, or it doesn't
-        # make sense, because we're not making a locale-specific request or the locale we're
-        # requesting is already the default locale.
+    if not (locale := kwargs.get("locale")):
         raise Http404
 
-    # We couldn't find a visible translation in the requested non-default locale, so let's
-    # see if we can find a visible translation via its parent.
+    slug = kwargs.get("slug")
+
+    if slug and look_for_translation_via_parent:
+        # Find documents with this slug in other locales
+        other_docs_qs = (
+            Document.objects.filter(slug=slug).exclude(locale=locale).select_related("parent")
+        )
+
+        if locale == settings.WIKI_DEFAULT_LANGUAGE:
```

Comment:
This feels inefficient. We have basically 2 blocks of code (if/else). Each block has the same loop and a call to `translated_to`. This should have been the other way. Loop once over the docs and check what you need for each one